From 1c72b17f9e000e5efa9ff71d6e07d912a2ac7539 Mon Sep 17 00:00:00 2001
From: ToLUSe <tobias@toluse.de>
Date: Mon, 22 Oct 2018 11:15:29 +0000
Subject: Add comment to Mali400 device tree config

---
 arch/arm/boot/dts/sun7i-a20.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index b4346e2e..d6834e91 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
+++ b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -1249,6 +1249,7 @@
 		mali: gpu@1c40000 {
 			compatible = "allwinner,sun7i-a20-mali", "arm,mali-400";
 			reg = <0x01c40000 0x10000>;
+			/* A20 datasheet 101-107 minus GIC_SPI offset of 32 */
 			interrupts = <GIC_SPI 69 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 70 IRQ_TYPE_LEVEL_HIGH>,
 				     <GIC_SPI 71 IRQ_TYPE_LEVEL_HIGH>,
-- 
2.19.1

