From c11afa6f9ccb7076ead7c148d460e89d78fc578a Mon Sep 17 00:00:00 2001
From: ToLuSe <tobias@toluse.de>
Date: Mon, 22 Oct 2018 11:15:29 +0000
Subject: Add Cubiescreen LCD to simple display

---
 drivers/gpu/drm/panel/panel-simple.c | 29 ++++++++++++++++++++++++++++
 1 file changed, 29 insertions(+)

diff --git a/drivers/gpu/drm/panel/panel-simple.c b/drivers/gpu/drm/panel/panel-simple.c
index 97964f7f..ada889c3 100644
--- a/drivers/gpu/drm/panel/panel-simple.c
+++ b/drivers/gpu/drm/panel/panel-simple.c
@@ -2323,6 +2323,32 @@ static const struct panel_desc winstar_wf35ltiacd = {
 	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
 };
 
+// CONFIG_VIDEO_LCD_MODE="x:800,y:480,depth:24,pclk_khz:27000,le:49,ri:70,up:2,lo:5,hs:1,vs:1,sync:3,vmode:0"
+static const struct drm_display_mode cubiescreen_mode = {
+	.clock = 27000,			// Clock cycle in KHz
+	.hdisplay = 800,		// Horizontal display period
+	.hsync_start = 800 + 50,	// 800 + 70
+	.hsync_end = 800 + 50 + 20,	// 800 + 70 + 1
+	.htotal = 920,			// Horizontal cycle
+	.vdisplay = 480,		// Vertical display period
+	.vsync_start = 480 + 3,		// 480 + 5
+	.vsync_end = 480 + 3 + 2,	// 480 + 5 + 1
+	.vtotal = 488,			// Vertical cycle
+	.vrefresh = 60,			// Vsync cycle
+	.flags = DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC, // Bug in sunxi-lcd? Display uses negative hsync and vsync
+};
+
+static const struct panel_desc cubiescreen = {
+	.modes = &cubiescreen_mode,
+	.num_modes = 1,
+	.bpc = 8,
+	.size = {
+		.width = 95,
+		.height = 54,
+	},
+	.bus_format = MEDIA_BUS_FMT_RGB888_1X24,
+};
+
 static const struct of_device_id platform_of_match[] = {
 	{
 		.compatible = "ampire,am-480272h3tmqw-t01h",
@@ -2573,6 +2599,9 @@ static const struct of_device_id platform_of_match[] = {
 	}, {
 		.compatible = "winstar,wf35ltiacd",
 		.data = &winstar_wf35ltiacd,
+	}, {
+		.compatible = "cubiescreen,lcd-cubiescreen",
+		.data = &cubiescreen,
 	}, {
 		/* sentinel */
 	}
-- 
2.19.1

