diff --git a/arch/arm/boot/dts/sun7i-a20.dtsi b/arch/arm/boot/dts/sun7i-a20.dtsi
index 4d63191cd317..68a3197ee3c4 100644
--- a/arch/arm/boot/dts/sun7i-a20.dtsi
+++ b/arch/arm/boot/dts/sun7i-a20.dtsi
@@ -103,11 +103,20 @@
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <0>;
-			clock-frequency = <912000000>;
+			clock-frequency = <960000000>;
 			clocks = <&ccu CLK_CPU>;
 			clock-latency = <244144>; /* 8 32k periods */
 			operating-points = <
 				/* kHz	  uV */
+				1488000	1960000
+				1440000	1880000
+				1392000	1800000
+				1344000	1720000
+				1296000	1650000
+				1248000	1600000
+				1200000	1525000
+				1152000	1510000
+				1008000	1450000
 				960000	1400000
 				912000	1400000
 				864000	1300000
@@ -123,11 +132,20 @@
 			compatible = "arm,cortex-a7";
 			device_type = "cpu";
 			reg = <1>;
-			clock-frequency = <912000000>;
+			clock-frequency = <960000000>;
 			clocks = <&ccu CLK_CPU>;
 			clock-latency = <244144>; /* 8 32k periods */
 			operating-points = <
 				/* kHz	  uV */
+				1488000	1960000
+				1440000	1880000
+				1392000	1800000
+				1344000	1720000
+				1296000	1650000
+				1248000	1600000
+				1200000	1525000
+				1152000	1510000
+				1008000	1450000
 				960000	1400000
 				912000	1400000
 				864000	1300000
