From 4279ce0820093dfab7656cf3bc1764514be3ae62 Mon Sep 17 00:00:00 2001
From: root <root@build.fritz.box>
Date: Fri, 19 Apr 2019 20:13:42 +0000
Subject: [PATCH] Added openvario.dts to Makefile

---
 arch/arm/dts/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 31d9256..730ece5 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -340,7 +340,8 @@ dtb-$(CONFIG_MACH_SUN7I) += \
 	sun7i-a20-primo73.dtb \
 	sun7i-a20-wexler-tab7200.dtb \
 	sun7i-a20-wits-pro-a20-dkt.dtb \
-	sun7i-a20-yones-toptech-bd1078.dtb
+	sun7i-a20-yones-toptech-bd1078.dtb \
+	openvario.dtb
 dtb-$(CONFIG_MACH_SUN8I_A23) += \
 	sun8i-a23-evb.dtb \
 	sun8i-a23-gt90h-v4.dtb \
-- 
1.8.3.1

