diff --git a/samples/auto.master b/samples/auto.master
index 9a9968a..cd861e8 100644
--- a/samples/auto.master
+++ b/samples/auto.master
@@ -4,13 +4,13 @@
 # mount-point [map-type[,format]:]map [options]
 # For details of the format look at auto.master(5).
 #
-/misc	/etc/auto.misc
+# /misc	/etc/auto.misc
 #
 # NOTE: mounts done from a hosts map will be mounted with the
 #	"nosuid" and "nodev" options unless the "suid" and "dev"
 #	options are explicitly given.
 #
-/net	-hosts
+# /net	-hosts
 #
 # Include /etc/auto.master.d/*.autofs
 # To add an extra map using this mechanism you will need to add
