From d9b43d42cfa6fa38c8bd2a76c16e173a27bced49 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Mon, 1 Sep 2014 06:37:18 +0200
Subject: [PATCH] Adapted toolchain prefixes for cross-compile

---
 build/compile.mk | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/build/compile.mk b/build/compile.mk
index 7b234ef..2a26172 100644
--- a/build/compile.mk
+++ b/build/compile.mk
@@ -6,28 +6,13 @@ ifeq ($(USE_CCACHE),y)
 endif
 
 EXE := $(findstring .exe,$(MAKE))
-AR = $(TCPREFIX)ar$(EXE)
-AS = $(TCPREFIX)as$(EXE)
-ifneq ($(ANALYZER),y)
-  ifeq ($(CLANG),y)
-    CXX = $(LLVM_PREFIX)clang++$(LLVM_SUFFIX)$(EXE)
-    CC = $(LLVM_PREFIX)clang$(LLVM_SUFFIX)$(EXE)
-  else
-    CXX = $(TCPREFIX)g++$(TCSUFFIX)$(EXE)
-    CC = $(TCPREFIX)gcc$(TCSUFFIX)$(EXE)
-  endif
-endif
-LD = $(TCPREFIX)ld$(EXE)
 DLLTOOL = $(TCPREFIX)dlltool$(EXE)
-SIZE = $(TCPREFIX)size$(EXE)
-STRIP = $(TCPREFIX)strip$(EXE)
 WINDRES = $(TCPREFIX)windres$(EXE)
 ARFLAGS = -rcs
 
 ifeq ($(TARGET),WINE)
 AR = ar$(EXE)
 LD = ld$(EXE) -m elf_i386
-STRIP = strip$(EXE)
 WINDRES = wrc$(EXE)
 endif
 
-- 
1.9.1

