From 7a975c59552c1a94cb0dd78cf2afa044ab412274 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Fri, 8 May 2015 20:20:05 +0200
Subject: [PATCH] Adapted toolchain prefix

---
 build/compile.mk | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/build/compile.mk b/build/compile.mk
index 7269752..d397c0b 100644
--- a/build/compile.mk
+++ b/build/compile.mk
@@ -6,39 +6,8 @@ ifeq ($(USE_CCACHE),y)
 endif
 
 EXE := $(findstring .exe,$(MAKE))
-AR = $(TCPREFIX)ar$(EXE)
-
-ifneq ($(ANALYZER),y)
-  ifeq ($(CLANG),y)
-    CXX = $(LLVM_PREFIX)clang++$(LLVM_SUFFIX)$(EXE)
-    CC = $(LLVM_PREFIX)clang$(LLVM_SUFFIX)$(EXE)
-  else
-    CXX = $(TCPREFIX)g++$(TCSUFFIX)$(EXE)
-    CC = $(TCPREFIX)gcc$(TCSUFFIX)$(EXE)
-  endif
-endif
-
-ifeq ($(CLANG),y)
-  AS = $(CC)
-  ASFLAGS += -c -xassembler
-  ifneq ($(LLVM_TARGET),)
-    ASFLAGS += -target $(LLVM_TARGET)
-  else
-    ASFLAGS += $(TARGET_ARCH)
-  endif
-
-  ifeq ($(MIPS),y)
-    # work around "Fatal error: invalid -march= option: `mips32'"
-    ASFLAGS += -integrated-as
-  endif
-else
-  AS = $(TCPREFIX)as$(EXE)
-endif
 
-LD = $(TCPREFIX)ld$(EXE)
 DLLTOOL = $(TCPREFIX)dlltool$(EXE)
-SIZE = $(TCPREFIX)size$(EXE)
-STRIP = $(TCPREFIX)strip$(EXE)
 WINDRES = $(TCPREFIX)windres$(EXE)
 ARFLAGS = -rcs
 
-- 
1.9.1

