From 79e11979652129e74386236ec5b3554a174d62e7 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <openvario@timoshome.org>
Date: Sun, 12 May 2019 12:15:15 +0000
Subject: [PATCH] Environment-Openvario-mainline

Upstream-Status: Inappropriate [embedded specific]

---
 include/configs/sunxi-common.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index d2d70f0fc2..594c512e3c 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -286,8 +286,8 @@
 
 #ifdef CONFIG_VIDEO
 #define CONSOLE_STDOUT_SETTINGS \
-	"stdout=serial,vidconsole\0" \
-	"stderr=serial,vidconsole\0"
+	"stdout=serial\0" \
+	"stderr=serial\0"
 #else
 #define CONSOLE_STDOUT_SETTINGS \
 	"stdout=serial\0" \
