From 74ae8499d7ab0a25f7617686e03ae6dec6decb6c Mon Sep 17 00:00:00 2001
From: Timo Bruderek <openvario@timoshome.org>
Date: Sun, 12 May 2019 12:15:15 +0000
Subject: [PATCH 1/1] Environment-Openvario-mainline

---
 include/configs/sunxi-common.h | 68 +++++++++++++++++++++++++++++++++++++++---
 1 file changed, 64 insertions(+), 4 deletions(-)

diff --git a/include/configs/sunxi-common.h b/include/configs/sunxi-common.h
index 9e37e99684..8d08a49004 100644
--- a/include/configs/sunxi-common.h
+++ b/include/configs/sunxi-common.h
@@ -403,8 +403,8 @@ extern int soft_i2c_gpio_scl;
 
 #ifdef CONFIG_DM_VIDEO
 #define CONSOLE_STDOUT_SETTINGS \
-	"stdout=serial,vidconsole\0" \
-	"stderr=serial,vidconsole\0"
+	"stdout=serial\0" \
+	"stderr=serial\0"
 #else
 #define CONSOLE_STDOUT_SETTINGS \
 	"stdout=serial\0" \
