From c793f27279ee4f32ebb26d85c9df086e3c61d00f Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo@timoshome.org>
Date: Fri, 23 Sep 2016 17:53:17 +0200
Subject: [PATCH 1/1] Add openvario.dts to Makefile

---
 arch/arm/dts/Makefile | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index ef573ec..ceb3e3f 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -223,7 +223,8 @@ dtb-$(CONFIG_MACH_SUN7I) += \
 	sun7i-a20-primo73.dtb \
 	sun7i-a20-wexler-tab7200.dtb \
 	sun7i-a20-wits-pro-a20-dkt.dtb \
-	sun7i-a20-yones-toptech-bd1078.dtb
+	sun7i-a20-yones-toptech-bd1078.dtb \
+	openvario.dtb
 dtb-$(CONFIG_MACH_SUN8I_A23) += \
 	sun8i-a23-evb.dtb \
 	sun8i-a23-gt90h-v4.dtb \
-- 
2.7.4

