From fc0e5854b02d191fadffa46c2763e402bb997788 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Sun, 7 Sep 2014 15:02:24 +0200
Subject: [PATCH] Override detection of target hardware

---
 build/detect.mk | 31 ++++++++++++++++---------------
 1 file changed, 16 insertions(+), 15 deletions(-)

diff --git a/build/detect.mk b/build/detect.mk
index 0a5953e..bb12a23 100644
--- a/build/detect.mk
+++ b/build/detect.mk
@@ -10,18 +10,18 @@ HOST_IS_MINGW := $(call string_contains,$(UNAME_S),MINGW)
 HOST_IS_WIN32 := $(call bool_or,$(HOST_IS_CYGWIN),$(HOST_IS_MINGW))
 HOST_IS_UNIX := $(call bool_not,$(HOST_IS_WIN32))
 
-HOST_IS_X86_32 := $(call bool_or,$(call string_contains,$(UNAME_M),i386),$(call string_contains,$(UNAME_M),i686))
-HOST_IS_X86_64 := $(call bool_or,$(call string_contains,$(UNAME_M),x86_64),$(call string_contains,$(UNAME_M),amd64))
 
-HOST_IS_ARM := $(call string_contains,$(UNAME_M),armv)
-HOST_IS_ARMV6 := $(call string_equals,$(UNAME_M),armv6l)
-HOST_IS_ARMV7 := $(call string_equals,$(UNAME_M),armv7l)
+#HOST_IS_ARM := $(call string_contains,$(UNAME_M),armv)
+#HOST_IS_ARMV6 := $(call string_equals,$(UNAME_M),armv6l)
+#HOST_IS_ARMV7 := $(call string_equals,$(UNAME_M),armv7l)
+HOST_IS_ARMV7 := y
 
-ifeq ($(HOST_IS_ARMV7),y)
-HOST_HAS_NEON := $(call string_contains,$(shell grep -E ^Features /proc/cpuinfo),neon)
-else
-HOST_HAS_NEON := n
-endif
+#ifeq ($(HOST_IS_ARMV7),y)
+#HOST_HAS_NEON := $(call string_contains,$(shell grep -E ^Features /proc/cpuinfo),neon)
+#else
+#HOST_HAS_NEON := n
+#endif
+HOST_HAS_NEON := y
 
 ifeq ($(HOST_IS_LINUX)$(HOST_IS_ARMV6),yy)
 # Check for VideoCore headers present on a Raspberry Pi
@@ -30,8 +30,9 @@ else
 HOST_IS_PI := n
 endif
 
-ifeq ($(HOST_IS_LINUX)$(HOST_IS_ARMV7),yy)
-HOST_HAS_MALI := $(call string_equals,$(shell test -c /dev/mali && echo y),y)
-else
-HOST_HAS_MALI := n
-endif
+#ifeq ($(HOST_IS_LINUX)$(HOST_IS_ARMV7),yy)
+#HOST_HAS_MALI := $(call string_equals,$(shell test -c /dev/mali && echo y),y)
+#else
+#HOST_HAS_MALI := n
+#endif
+HOST_HAS_MALI := y
-- 
1.9.1

