--- a/src/Form/ButtonPanel.cpp
+++ b/src/Form/ButtonPanel.cpp
@@ -353,7 +353,7 @@ ButtonPanel::KeyPress(unsigned key_code)
     }
   }
 
-  if (selected_index >= 0 && !HasPointer()) {
+  if (selected_index >= 0) {
     if (key_code == KEY_LEFT) {
       SelectPrevious();
       return true;
       
--- a/src/Form/Button.cpp
+++ b/src/Form/Button.cpp
@@ -239,7 +239,7 @@ Button::OnPaint(Canvas &canvas)
 
   const bool pressed = down;
   const bool focused = HasCursorKeys()
-    ? HasFocus() || (selected && !HasPointer())
+    ? HasFocus() || (selected)
     : pressed;
 
   renderer->DrawButton(canvas, GetClientRect(),
