From 811594fcc643621f1614932249788727d64325e4 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Mon, 1 Sep 2014 06:38:46 +0200
Subject: [PATCH] Adapted Flags for compiler and linker for cross compile

---
 build/targets.mk | 13 -------------
 1 file changed, 13 deletions(-)

diff --git a/build/targets.mk b/build/targets.mk
index a172f7a..ded1979 100644
--- a/build/targets.mk
+++ b/build/targets.mk
@@ -463,11 +463,6 @@ ifeq ($(HOST_IS_PI)$(TARGET_IS_PI),ny)
   TARGET_CPPFLAGS += --sysroot=$(PI) -isystem $(PI)/usr/include/arm-linux-gnueabihf
 endif
 
-ifeq ($(HOST_IS_ARM)$(TARGET_HAS_MALI),ny)
-  # cross-crompiling for Cubieboard
-  TARGET_CPPFLAGS += --sysroot=$(CUBIE) -isystem $(CUBIE)/usr/include/arm-linux-gnueabihf
-  TARGET_CPPFLAGS += -isystem $(CUBIE)/usr/local/stow/sunxi-mali/include
-endif
 
 ifeq ($(TARGET_IS_KOBO),y)
   TARGET_CPPFLAGS += -DKOBO
@@ -549,14 +544,6 @@ ifeq ($(HOST_IS_PI)$(TARGET_IS_PI),ny)
   TARGET_LDFLAGS += --sysroot=$(PI) -L$(PI)/usr/lib/arm-linux-gnueabihf
 endif
 
-ifeq ($(HOST_IS_ARM)$(TARGET_HAS_MALI),ny)
-  # cross-crompiling for Cubieboard
-  TARGET_LDFLAGS += --sysroot=$(CUBIE)
-  TARGET_LDFLAGS += -L$(CUBIE)/lib/arm-linux-gnueabihf
-  TARGET_LDFLAGS += -L$(CUBIE)/usr/lib/arm-linux-gnueabihf
-  TARGET_LDFLAGS += -L$(CUBIE)/usr/local/stow/sunxi-mali/lib
-endif
-
 ifeq ($(TARGET_IS_KOBO),y)
   TARGET_LDFLAGS += -L$(KOBO)/lib
   TARGET_LDFLAGS += -static-libstdc++
-- 
1.9.1

