From e7254802e4e357754125568f9fa3e8fa78ff5068 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <openvario@timoshome.org>
Date: Sat, 28 Sep 2019 12:21:14 +0000
Subject: [PATCH] Driver OpenVario: Add POV V NMEA sentence

---
 src/Device/Driver/OpenVario.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/Device/Driver/OpenVario.cpp b/src/Device/Driver/OpenVario.cpp
index 68edaf9..4b33e33 100644
--- a/src/Device/Driver/OpenVario.cpp
+++ b/src/Device/Driver/OpenVario.cpp
@@ -108,6 +108,11 @@ OpenVarioDevice::POV(NMEAInputLine &line, NMEAInfo &info)
         info.temperature_available = true;
         break;
       }
+      case 'V': {
+        info.voltage = value;
+        info.voltage_available.Update(info.clock);
+        break;
+      }
     }
   }
 
-- 
1.8.3.1

