From c1ef1df21887c99771f0524b2a8cc588d6635a61 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Wed, 6 Jan 2016 21:28:52 +0100
Subject: [PATCH] Workaround for lua detection

---
 build/lua.mk | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/build/lua.mk b/build/lua.mk
index 17110fb..a77fff9 100644
--- a/build/lua.mk
+++ b/build/lua.mk
@@ -2,12 +2,8 @@ LUA = y
 
 ifeq ($(LUA),y)
 
-ifeq ($(USE_THIRDARTY_LIBS),y)
 LUA_LDLIBS = -llua
 LUA_CPPFLAGS =
-else
-$(eval $(call pkg-config-library,LUA,lua5.2))
-endif
 
 LUA_SOURCES = \
 	$(SRC)/Lua/Error.cpp \
-- 
1.9.1

