From 2d1994d5ecf1b7d7fdfee403631c8f2a65b7d87d Mon Sep 17 00:00:00 2001
From: bomilkar <ronald_niederhagen@freenet.de>
Date: Thu, 6 Jun 2024 13:57:15 +0200
Subject: [PATCH 2/3] replace defaulted equality comparison

---
 src/Geo/Flat/FlatGeoPoint.hpp | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/src/Geo/Flat/FlatGeoPoint.hpp b/src/Geo/Flat/FlatGeoPoint.hpp
index db30016756..5f69b8dc24 100644
--- a/src/Geo/Flat/FlatGeoPoint.hpp
+++ b/src/Geo/Flat/FlatGeoPoint.hpp
@@ -69,7 +69,9 @@ struct FlatGeoPoint : IntPoint2D {
     return ::DotProduct(*this, other);
   }
 
-  constexpr bool operator==(const FlatGeoPoint &) const noexcept = default;
+  constexpr bool operator==(const FlatGeoPoint &other) const noexcept {
+    return IntPoint2D::operator==(other);
+  }
 
   [[gnu::pure]]
   bool Sort(const FlatGeoPoint& sp) const noexcept {
@@ -106,7 +108,9 @@ struct AFlatGeoPoint : public FlatGeoPoint {
     y = (y >> 2) << 2;
   }
 
-  constexpr bool operator==(const AFlatGeoPoint &) const noexcept = default;
+  constexpr bool operator==(const AFlatGeoPoint &other) const noexcept {
+    return FlatGeoPoint::operator==(other) && (altitude == other.altitude);
+  }
 
   /**
    * Ordering operator, used for set ordering.  Uses lexicographic comparison.
-- 
2.39.2

