From b720859b8c525d6a9c70b4bfd08d27dae0fb0008 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Tue, 3 May 2016 21:49:12 +0200
Subject: [PATCH 1/1] Workaround for lua detection

---
 build/lua.mk | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/build/lua.mk b/build/lua.mk
index 4ef5276..aadb7ef 100644
--- a/build/lua.mk
+++ b/build/lua.mk
@@ -2,12 +2,8 @@ LUA = y
 
 ifeq ($(LUA),y)
 
-ifeq ($(USE_THIRDPARTY_LIBS),y)
 LIBLUA_LDLIBS = -llua
 LIBLUA_CPPFLAGS =
-else
-$(eval $(call pkg-config-library,LIBLUA,lua5.2))
-endif
 
 LUA_SOURCES = \
 	$(SRC)/Lua/Error.cpp \
-- 
1.9.1

