From e0df67faa1e8fccf633655c685309ebc8110a792 Mon Sep 17 00:00:00 2001
From: Timo Bruderek <timo.bruderek@gmx.de>
Date: Wed, 19 Aug 2015 17:21:35 +0200
Subject: [PATCH] Addapt cross compile toolchain

---
 build/compile.mk | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/build/compile.mk b/build/compile.mk
index d0dad76..86ef6e4 100644
--- a/build/compile.mk
+++ b/build/compile.mk
@@ -6,39 +6,8 @@ ifeq ($(USE_CCACHE),y)
 endif
 
 EXE := $(findstring .exe,$(MAKE))
-AR = $(TCPREFIX)ar$(EXE)
-
-ifneq ($(ANALYZER),y)
-  ifeq ($(CLANG),y)
-    CXX = $(LLVM_PREFIX)clang++$(LLVM_SUFFIX)$(EXE)
-    CC = $(LLVM_PREFIX)clang$(LLVM_SUFFIX)$(EXE)
-  else
-    CXX = $(TCPREFIX)g++$(TCSUFFIX)$(EXE)
-    CC = $(TCPREFIX)gcc$(TCSUFFIX)$(EXE)
-  endif
-endif
-
-ifeq ($(CLANG),y)
-  AS = $(CC)
-  ASFLAGS += -c -xassembler
-  ifneq ($(LLVM_TARGET),)
-    ASFLAGS += -target $(LLVM_TARGET)
-  else
-    ASFLAGS += $(TARGET_ARCH)
-  endif
-
-  ifeq ($(call bool_or,$(MIPS),$(MIPS64)),y)
-    # work around "Fatal error: invalid -march= option: `mips32'"
-    ASFLAGS += -integrated-as
-  endif
-else
-  AS = $(TCPREFIX)as$(EXE)
-endif
 
-LD = $(TCPREFIX)ld$(EXE)
 DLLTOOL = $(TCPREFIX)dlltool$(EXE)
-SIZE = $(TCPREFIX)size$(EXE)
-STRIP = $(TCPREFIX)strip$(EXE)
 WINDRES = $(TCPREFIX)windres$(EXE)
 ARFLAGS = -rcs
 
-- 
1.9.1

