From 7402120441731ad8832086c711b48fe0cfa53060 Mon Sep 17 00:00:00 2001
From: root <root@ov-build-092016-gcc52.gallien>
Date: Mon, 9 Jan 2017 20:42:51 +0100
Subject: [PATCH 1/1] Adapted toolchain prefixes for cross compile

---
 build/compile.mk | 32 --------------------------------
 1 file changed, 32 deletions(-)

diff --git a/build/compile.mk b/build/compile.mk
index 8acc53e..c11bee3 100644
--- a/build/compile.mk
+++ b/build/compile.mk
@@ -6,40 +6,8 @@ ifeq ($(USE_CCACHE),y)
 endif
 
 EXE := $(findstring .exe,$(MAKE))
-AR = $(TCPREFIX)ar$(EXE)
-RANLIB = $(TCPREFIX)ranlib$(EXE)
-
-ifneq ($(ANALYZER),y)
-  ifeq ($(CLANG),y)
-    CXX = $(LLVM_PREFIX)clang++$(LLVM_SUFFIX)$(EXE)
-    CC = $(LLVM_PREFIX)clang$(LLVM_SUFFIX)$(EXE)
-  else
-    CXX = $(TCPREFIX)g++$(TCSUFFIX)$(EXE)
-    CC = $(TCPREFIX)gcc$(TCSUFFIX)$(EXE)
-  endif
-endif
-
-ifeq ($(CLANG),y)
-  AS = $(CC)
-  ASFLAGS += -c -xassembler
-  ifneq ($(LLVM_TARGET),)
-    ASFLAGS += -target $(LLVM_TARGET)
-  else
-    ASFLAGS += $(TARGET_ARCH)
-  endif
-
-  ifeq ($(call bool_or,$(MIPS),$(MIPS64)),y)
-    # work around "Fatal error: invalid -march= option: `mips32'"
-    ASFLAGS += -integrated-as
-  endif
-else
-  AS = $(TCPREFIX)as$(EXE)
-endif
 
-LD = $(TCPREFIX)ld$(EXE)
 DLLTOOL = $(TCPREFIX)dlltool$(EXE)
-SIZE = $(TCPREFIX)size$(EXE)
-STRIP = $(TCPREFIX)strip$(EXE)
 WINDRES = $(TCPREFIX)windres$(EXE)
 ARFLAGS = -rcs
 
-- 
1.8.3.1

