diff --git a/build/compile.mk b/build/compile.mk
index 91d069bfc8..e1af770963 100644
--- a/build/compile.mk
+++ b/build/compile.mk
@@ -6,40 +6,7 @@ ifeq ($(USE_CCACHE),y)
 endif
 
 EXE := $(findstring .exe,$(MAKE))
-AR = $(TCPREFIX)ar$(EXE)
-RANLIB = $(TCPREFIX)ranlib$(EXE)
-
-ifneq ($(ANALYZER),y)
-  ifeq ($(CLANG),y)
-    CXX = $(LLVM_PREFIX)clang++$(LLVM_SUFFIX)$(EXE)
-    CC = $(LLVM_PREFIX)clang$(LLVM_SUFFIX)$(EXE)
-  else
-    CXX = $(TCPREFIX)g++$(TCSUFFIX)$(EXE)
-    CC = $(TCPREFIX)gcc$(TCSUFFIX)$(EXE)
-  endif
-endif
-
-ifeq ($(CLANG),y)
-  AS = $(CC)
-  ASFLAGS += -c -xassembler
-  ifneq ($(LLVM_TARGET),)
-    ASFLAGS += -target $(LLVM_TARGET)
-  else
-    ASFLAGS += $(TARGET_ARCH)
-  endif
-
-  ifeq ($(call bool_or,$(MIPS),$(MIPS64)),y)
-    # work around "Fatal error: invalid -march= option: `mips32'"
-    ASFLAGS += -integrated-as
-  endif
-else
-  AS = $(TCPREFIX)as$(EXE)
-endif
-
-LD = $(TCPREFIX)ld$(EXE)
 DLLTOOL = $(TCPREFIX)dlltool$(EXE)
-SIZE = $(TCPREFIX)size$(EXE)
-STRIP = $(TCPREFIX)strip$(EXE)
 WINDRES = $(TCPREFIX)windres$(EXE)
 ARFLAGS = rcs
 
